perfkitbenchmarker/linux_benchmarks/iperf_benchmark.py

```diff
@@ -98,63 +126,191 @@ def _RunIperf(sending_vm, receiving_vm, receiving_ip_address, thread_count,
+        raise ValueError(f'Only {len(thread_values)} out of {thread_count}'
+                         ' iperf threads reported a throughput value.')
+
+    total_throughput = 0.0
+    for value in thread_values:
+      total_throughput += float(value)
+
+    metadata = {
+        # The meta data defining the environment
+        'receiving_machine_type': receiving_vm.machine_type,
+        'receiving_zone': receiving_vm.zone,
+        'sending_machine_type': sending_vm.machine_type,
+        'sending_thread_count': thread_count,
+        'sending_zone': sending_vm.zone,
+        'runtime_in_seconds': FLAGS.iperf_runtime_in_seconds,
+        'ip_type': ip_type,
+        'buffer_size': buffer_size,
+        'tcp_window_size': window_size,
+        'write': write,
+        'err': err,
+        'retry': retry,
+        'cwnd': cwnd,
+        'rtt': rtt,
+        'rtt_unit': rtt_unit,
+        'netpwr': netpwr
+    }
+    return sample.Sample('Throughput', total_throughput, 'Mbits/sec', metadata)
+
+  elif protocol == UDP:
+
+    iperf_cmd = ( f'iperf -e -u --client {receiving_ip_address} --port {IPERF_UDP_PORT} '
+                  f'--format m --time {FLAGS.iperf_runtime_in_seconds} -P {thread_count}')
+
+    if FLAGS.iperf_udp_per_stream_bandwidth:
+      iperf_cmd += f' -b {FLAGS.iperf_udp_per_stream_bandwidth}M'
+
+    # the additional time on top of the iperf runtime is to account for the
+    # time it takes for the iperf process to start and exit
+    timeout_buffer = FLAGS.iperf_timeout or 30 + thread_count
+    stdout, _ = sending_vm.RemoteCommand(iperf_cmd, should_log=True,
+                                         timeout=FLAGS.iperf_runtime_in_seconds +
+                                         timeout_buffer)
+
+    multi_thread = re.search((r'\[SUM\]\s\d+\.?\d+-\d+\.?\d+\ssec\s+\d+\.?\d+\s+MBytes\s+\d+\.?\d+'
+                              r'\s+Mbits\/sec\s+(?P<write>\d+)\/(?P<err>\d+)\s+(?P<pps>\d+)\s+pps'),
+                             stdout)
+
+    match = re.search(r'UDP buffer size: (?P<buffer_size>\d+\.\d+)\s+(?P<buffer_unit>\w+)', stdout)
+    buffer_size = float(match.group('buffer_size'))
+    buffer_size_unit = match.group('buffer_unit')
+    datagram_size = int(re.findall(r'(\d+)\sbyte\sdatagrams', stdout)[0])
+    ipg_target = float(re.findall(r'IPG\starget:\s(\d+.?\d+)', stdout)[0])
+    ipg_target_unit = str(re.findall(r'IPG\starget:\s\d+.?\d+\s(\S+)\s', stdout)[0])
+
+    if multi_thread:
+      # Write, Err, PPS
+      write = int(multi_thread.group('write'))
+      err = int(multi_thread.group('err'))
+      pps = int(multi_thread.group('pps'))
+
+    else:
+      # Write, Err, PPS
+      match = re.search(r'\d+\s+Mbits\/sec\s+(?P<write>\d+)\/(?P<err>\d+)\s+(?P<pps>\d+)\s+pps', stdout)
+      write = int(match.group('write'))
+      err = int(match.group('err'))
+      pps = int(match.group('pps'))
+
+    # Jitter
+    jitter_array = re.findall(r'Mbits\/sec\s+(\d+\.?\d+)\s+[a-zA-Z]+', stdout)
```

Comment:
You shouldn't have to escape forward slashes.  Instead of "\/" just have "/" to improve readability